src/test/java/StringManipulationTest.java

```diff
@@ -20,66 +27,93 @@ public void tearDown() {
-        fail("Not yet implemented");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
     @Test
+    //this test check for correct output
     public void testCount4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("hunkady hunkad hunka hunk hun hu h");
+        int length = manipulatedstring.count();
+        assertEquals(7, length);
     }
 
     @Test
-    public void testRemoveNthCharacter1() {
+    //this test check for correct output
+    public void testRemoveNthCharacter1() throws Exception {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
     }
 
     @Test
-    public void testRemoveNthCharacter2() {
+    //this test check for correct output
+    public void testRemoveNthCharacter2() throws Exception {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
     }
 
     @Test
-    public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+    //this test check for correct output by leaving the string with its default value
+    public void testRemoveNthCharacter3() throws Exception {
+        assertEquals(null, manipulatedstring.removeNthCharacter(3, false));
     }
 
     @Test
-    public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+    //this test check for correct output by giving the string an empty string value
+    public void testRemoveNthCharacter4() throws Exception {
+        manipulatedstring.setString("");
+        assertEquals("", manipulatedstring.removeNthCharacter(4, true));
     }
 
     @Test
-    public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+    //this test checks to see if the IllegalArgumentException is thrown
+    public void testRemoveNthCharacter5() throws Exception {
+        manipulatedstring.setString("Hi Sara!");
+
+        assertThrows(IllegalArgumentException.class, () -> {
+            manipulatedstring.removeNthCharacter(-1, true);
+        });
     }
 
     @Test
+    //this class checks to see if the IndexOutOfBoundsException is thrown
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hi Sara!");
+
+        assertThrows(IndexOutOfBoundsException.class, () -> {
+            manipulatedstring.removeNthCharacter(100, true);
+        });
     }
 
     @Test
-    public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+    //this test check for correct output
+    public void testRemoveNthCharacter7() throws Exception {
+        manipulatedstring.setString("CS4420");
+        assertEquals("CS420", manipulatedstring.removeNthCharacter(4, false));
     }
 
```

Comment:
One note on the getSubstrings() method:
I notice that you don't have test cases that include punctuation, like "I need some sleep!". The reason I mention this is that I can tell by the way you split strings, the "!" will be included with the string. I'd recommend using a regex to extract your words, or additional processing on the temp array you use to hold strings to remove punctuations.